app/controllers/checkout_controller.rb

```diff
@@ -0,0 +1,54 @@
+class CheckoutController < ApplicationController
+  include Wicked::Wizard
+  attr_reader :step
+
+  authorize_resource class: false
+
+  before_action :set_order
+
+  steps :login, :addresses, :delivery, :payment, :confirm, :complete
+
+  def show
+    return redirect_to root_path if session[:line_item_ids].nil? && step == :addresses
+
+    step == :login ? login : handle_show_step
+    render_wizard
+  end
+
+  def update
+    handle_update_step
+    redirect_to next_wizard_path unless performed?
+  end
+
+  private
+
+  def handle_show_step
+    return jump_to(previous_step) if Checkout::ConditionStepService.new(current_user: current_user,
```

Comment:
A user should be authorized on this step. So we don`t need to handle the current user in this service

Reply:
U receive all session here but need just one arg of it